lib/ts/recipe/totp/types.ts

```diff
@@ -0,0 +1,200 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import type { BlockedScreen } from "./components/themes/mfa/blockedScreen";
+import type { LoadingScreen } from "./components/themes/mfa/loadingScreen";
+import type { CodeForm } from "./components/themes/mfa/totpCodeForm";
+import type { CodeVerificationFooter } from "./components/themes/mfa/totpCodeVerificationFooter";
+import type { CodeVerificationHeader } from "./components/themes/mfa/totpCodeVerificationHeader";
+import type { DeviceInfoSection } from "./components/themes/mfa/totpDeviceInfoSection";
+import type { DeviceSetupFooter } from "./components/themes/mfa/totpDeviceSetupFooter";
+import type { DeviceSetupHeader } from "./components/themes/mfa/totpDeviceSetupHeader";
+import type { ComponentOverride } from "../../components/componentOverride/componentOverride";
+import type { UserContext } from "../../types";
+import type {
+    Config as RecipeModuleConfig,
+    NormalisedConfig as NormalisedRecipeModuleConfig,
+    UserInput as RecipeModuleUserInput,
+} from "../recipeModule/types";
+import type { Dispatch } from "react";
+import type { OverrideableBuilder } from "supertokens-js-override";
+import type { RecipeInterface } from "supertokens-web-js/recipe/totp";
+
+export type ComponentOverrideMap = {
+    TOTPBlockedScreen_Override?: ComponentOverride<typeof BlockedScreen>;
+    TOTPLoadingScreen_Override?: ComponentOverride<typeof LoadingScreen>;
+    TOTPCodeForm_Override?: ComponentOverride<typeof CodeForm>;
```

Comment:
is this component reused in the enter totp UI and also the verify device part of the create device UI?

Reply:
it is.